content/blogposts/2024/improving-cody-autocomplete-faster-smarter.md

```diff
@@ -0,0 +1,132 @@
+    caption="Faster completions served from cache"
+/>
+
+### Implementing Smart Throttle
+
+Smart Throttle is a client-side optimization technique implemented in Cody to improve autocomplete responsiveness. It works by intelligently reducing the debounce interval between user input and completion requests.
+
+Unlike traditional fixed debounce timers, Smart Throttle dynamically adjusts its timing based on typing patterns and context. This allows for faster suggestion delivery without overwhelming the server with excessive requests.
+
+The implementation of Smart Throttle contributed significantly to Cody's performance improvements. It helped reduce overall client-side latency by about 110ms across all line-count groups. Additionally, it played a role in **increasing the number of suggested characters by 24% and boosting cache utilization by 35%**.
+
+## Smarter Suggestions: 58% More Accepted Code
+
+Not only is Cody faster, but it's also getting better at suggesting code you actually want to use. The DeepSeek v2 model has significantly improved the quality of suggestions. 
+
+We’ve seen a **58% increase** in the number of accepted characters per user. This means that the suggestions provided by Cody are more likely to be relevant and accepted. In practical terms, if you're writing a 1000-line program, Cody is now potentially writing 580 of those lines for you, up from about 367 before.
+
+- Suggested characters per user increased from 1,600 to 2,500.
+- Accepted characters per user increased from 600 to 950.
+
+Here’s our analytics tracking suggested characters and accepted characters:
+
+<Figure 
+    src="https://storage.googleapis.com/sourcegraph-assets/blog/faster-autocomplete/autocomplete-characters.png"
+    caption="Suggested and accepted characters per user"
+/>
+
+## Better Multi-Line Completions
+
+Our A/B testing showed significant improvements for multi-line completions, thanks to DeepSeek v2's enhanced ability to understand context:
+
+For 2-5 line completions:
+
+### Completion Acceptance Rate (CAR) improved by 4.24%
+
+CAR is the percentage of suggested completions that are accepted by the user. It's calculated by dividing the number of accepted completions by the total number of suggested completions. 
+
+An improvement of 4.24% means that users are accepting a higher proportion of Cody’s multi-line suggestions.
+
+### 30-second Persistence Rate improved by 2.53%
+
+A 30-second Persistence Rate specifically looks at whether the accepted completion is still present in the code 30 seconds after it was inserted. An improvement of 2.53% suggests that more of Cody's multi-line completions are being kept by users.
+
+## Improvements for Cody Enterprise Users
+
+For our Cody Enterprise users, we've also made significant improvements, though the exact numbers differ slightly from our PLG users due to the unique requirements of enterprise environments. 
```

Comment:
I think of "PLG" as an internal term. Users probably identify with their plan -- Free or Pro

```suggestion
For our Cody Enterprise users, we've also made significant improvements, though the exact numbers differ slightly from our Free and Pro users due to the unique requirements of enterprise environments. 
```